pkg/webhook/validate_service.go

```diff
@@ -46,10 +47,20 @@ const (
 func ValidateRevisionTemplate(ctx context.Context, uns *unstructured.Unstructured) error {
 	content := uns.UnstructuredContent()
 
+	var mode DryRunMode
+	features := config.FromContextOrDefaults(ctx).Features
+	switch features.PodSpecDryRun {
+	case config.Enabled:
+		mode = DryRunEnabled
+	case config.Disabled:
+		return nil
+	default:
+		mode = DryRunMode(uns.GetAnnotations()[PodSpecDryRunAnnotation])
```

Comment:
should this be same as disabled if annotation is not set?

Reply:
The feature is only client controlled by the annotation in "allowed" mode. Vendors may disable the feature completely or always run it with disable/allowed feature flags.